### PR TITLE
Feat/river charge

### PR DESCRIPTION
**Pull request by Immo-Be**
for #1604 

- adds a script to recursively download the `river discharge data`
- add Achim's script to parse single files

### CODE REVIEW

File: data/downloads/recursive-table-download.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    except requests.RequestException as e:
+        log_message(f"Failed to fetch {url}: {e}")
+        global had_failures
+        had_failures = True
+        return
+
+    soup = BeautifulSoup(response.content, "html.parser")
+    # Find the table containing the links
+    # This works because the tables containing the links have a the same specific class
+    table = soup.find("table", class_="table table-sm")
```
Comment (andreashelms) [inline data/downloads/recursive-table-download.py:37]: Interesting approach 🤔 Maybe it's easier and less error-prone if we use FTP within a shell script for downloading:
ftp://anon-ftp.ceda.ac.uk/neodc/esacci/river_discharge/data/WL/v1.1/CSV/merged_timeseries

I should have told you before that most datasets are also available via FTP.

File: data/downloads/parse-river-discharte-data.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+def read_altimetry_data(filename):
+    """Reads the custom CSV file, cleaning up header and separator."""
+    with open(filename, 'r') as f:
+        lines = f.readlines()
+
+    data_lines = [line.strip() for line in lines if not line.startswith("#")]
+    df = pd.read_csv(StringIO('\n'.join(data_lines)), sep=' ', header=None)
+    return df
+
+df = read_altimetry_data("./download/xco2/R_AMAZON_AMAZONA_KMXXXX_mergedjason3-0228_S0234 copy.csv")
```
Comment (andreashelms) [inline data/downloads/parse-river-discharte-data.py:16]: Can you update this to read in all the downloaded files?

File: data/downloads/generate-timestamp-river-discharge-data.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
+import os
+import pandas as pd
+import geopandas as gpd
+from io import StringIO
+import json
+
+# Directory containing the CSV files
+data_dir = "/data/downloads/download/xco2"
```
Comment (andreashelms) [inline data/downloads/generate-timestamp-river-discharge-data.py:8]: I think this needs to be changed

Comment (Immo-Be) [reply data/downloads/generate-timestamp-river-discharge-data.py:8]: To what?

Comment (andreashelms) [reply data/downloads/generate-timestamp-river-discharge-data.py:8]: e.g. "/data/downloads/download/river_discharge"

File: data/downloads/recursive-table-download.py
Comment (andreashelms) [inline data/downloads/recursive-table-download.py:1]: I would suggest to rename this to `neodc-river-discharge` follwing the convention of the other file namings

File: data/downloads/generate-timestamp-river-discharge-data.py
Comment (andreashelms) [inline data/downloads/generate-timestamp-river-discharge-data.py:1]: Please move this file to `data/`. `data/downloads/` should only contain the actual download scripts.

File: data/downloads/parse-river-discharge-data.py
Comment (andreashelms) [inline data/downloads/parse-river-discharge-data.py:1]: Same here. Please move this file to `data/`.

**Review by andreashelms — APPROVED** [review]

File: data/downloads/neodc-river-discharge.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import ftplib
+import os
+
+# Base FTP URL
+ftp_url = "anon-ftp.ceda.ac.uk"
+base_path = "/neodc/esacci/river_discharge/data/WL/v1.1/CSV/merged_timeseries/"
+
+# Directory to save the downloaded files
+download_dir = "download/xco2"
```
Comment (andreashelms) [inline data/downloads/neodc-river-discharge.py:9]:
```suggestion
download_dir = "download/river_discharge"
```

**Review by andreashelms — CHANGES_REQUESTED** [review]

File: data/parse-river-discharge-data.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
+import os
+import pandas as pd
+import geopandas as gpd
+from io import StringIO
+import json
+
+# Directory containing the CSV files
+data_dir = "/data/downloads/download/xco2"
```
Comment (andreashelms) [inline data/parse-river-discharge-data.py:8]:
```suggestion
data_dir = "/data/downloads/download/river_discharge"
```

**Review by andreashelms — APPROVED** [review]

**Pull request merged**